File: PT_Boldestio_Python/models/tour.py
from dataclasses import dataclass, field
from .day import Day
from typing import List

@dataclass
class Tour:
    externalID: str
    title_es: str
    htmlDescription_es: str
    htmlContent_es: str
    program: List[Day] = field(default_factory=list)

    def add_day(self, day: Day):
        # Añade una instancia de Day a la lista program
        if not isinstance(day, Day):
            raise TypeError("Expected an instance of Day")
        self.program.append(day)

File: PT_Boldestio_Java/src/main/java/script/Tour.java
package script;

import java.util.LinkedList;


public class Tour {

    //@CsvBindByPosition(position = 1)
    private String externalID;
    //@CsvBindByPosition(position = 4)
    private String title_es;
    //@CsvBindByPosition(position = 5)
    private String htmlDescription_es;
    //@CsvBindByPosition(position = 6)
    private String htmlContent_es;
    private LinkedList<Day> program;

    public Tour(String id, String title, String htmlDescription, String htmlContent) {

        this.externalID = id;
        this.title_es = title;
        this.htmlDescription_es = htmlDescription;
        this.htmlContent_es = htmlContent;
        this.program = new LinkedList<Day>();
    }

    public void annadirDay(Day day){
        this.program.add(day);
    }
}
